java-checks/src/main/java/org/sonar/java/checks/DefaultEncodingUsageCheck.java

```diff
@@ -164,7 +196,27 @@ private static MethodMatcher constructor(String type) {
 
   @Override
   protected void onMethodInvocationFound(MethodInvocationTree mit) {
-    reportIssue(MethodsHelper.methodName(mit), "Remove this use of \"" + mit.symbol().name() + "\"");
+    if (COMMONS_IO_CHARSET_MATCHERS.anyMatch(mit)) {
+      Arguments arguments = mit.arguments();
+      ExpressionTree lastArgument = arguments.get(arguments.size() - 1);
+      testNullLiteralPassedForEncoding(lastArgument);
+    } else if (FILEUTILS_WRITE_WITH_CHARSET_MATCHERS.anyMatch(mit)) {
+      testNullLiteralPassedForEncoding(mit.arguments().get(2));
+    } else {
+      reportIssue(MethodsHelper.methodName(mit), "Remove this use of \"" + mit.symbol().name() + "\"");
+    }
+  }
+
+  private void testNullLiteralPassedForEncoding(ExpressionTree argument) {
+    if (isNullLiteral(argument)) {
+      reportIssue(argument, "Replace this \"null\" with actual charset.");
+    }
+  }
+
+  private static boolean isNullLiteral(ExpressionTree lastArgument) {
+    ExpressionTree arg = ExpressionUtils.skipParentheses(lastArgument);
+    return arg.is(Tree.Kind.NULL_LITERAL)
+      || (arg.is(Tree.Kind.TYPE_CAST) && ((TypeCastTree) arg).expression().is(Tree.Kind.NULL_LITERAL));
```

Comment:
wonder if smart or over cautious but the last operand could be replaced by `isNullLiteral((TypeCastTree) arg).expression())`

And this should be tested with something like : `IOUtils.write(s, output, ((String) (null)));`